internal/querynodev2/segments/manager.go

```diff
@@ -165,37 +165,51 @@ func (mgr *segmentManager) Put(segmentType SegmentType, segments ...Segment) {
 	for _, segment := range segments {
 		oldSegment, ok := targetMap[segment.ID()]
 
-		if ok && oldSegment.Version() >= segment.Version() {
-			log.Warn("Invalid segment distribution changed, skip it",
-				zap.Int64("segmentID", segment.ID()),
-				zap.Int64("oldVersion", oldSegment.Version()),
-				zap.Int64("newVersion", segment.Version()),
-			)
-			continue
+		if ok {
+			if oldSegment.Version() >= segment.Version() {
+				log.Warn("Invalid segment distribution changed, skip it",
+					zap.Int64("segmentID", segment.ID()),
+					zap.Int64("oldVersion", oldSegment.Version()),
+					zap.Int64("newVersion", segment.Version()),
+				)
+				// delete redundant segment
+				if s, ok := segment.(*LocalSegment); ok {
+					DeleteSegment(s)
+				}
+				continue
+			}
+			replacedSegment = append(replacedSegment, oldSegment)
```

Comment:
prefer to remove this logic, any dup `Put` has to be rejected. The version should be only updated by leader observer.

Keep only less paths to reach updating version, keep the system simple

Reply:
Actually this won't happen as segment loader has filtered out the dup load request

Reply:
My proposal is, keeping this dup check, but remove the updating version logic